### ISSUE TITLE
Allow for custom hyperlinks for matched patterns

### ISSUE THREAD

**Issue by jaraco** [zeit/hyper#3851]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: macOS 10.15
- **Hyper.app version**: 3.0.2
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/4e4b1f6a93bd6ee16694d17ed80826c8
- **Relevant information from devtools** _(CMD+ALT+I on macOS, CTRL+SHIFT+I elsewhere)_: 
- **The issue is reproducible in vanilla Hyper.app**: `Is Vanilla`

## Issue

I'd like to be able to make arbitrary patterns hyperlinkable. Consider the [old plugin](https://github.com/zeit/hyperlinks), which seems to have recognized e-mail addressed and made them hyperlinkable. I'd like to do something similar with other patterns, such as with Python's bpo-NNNN (which would link to bugs.python.org/issues/NNNN).

I'd like to be able to achieve this while relying on the code within Hyper.js to render the links and handle the click events. Ideally, Hyper.js would offer a configuration parameter mapping regexes to their hyperlinks and allowing groups from the regexes to be specified in the resultant hyperlink. Extra credit if the rendering could also be a javascript function accepting the match object and returning the hyperlink.

**Comment by Stanzilla** [comment]
We are now using https://github.com/xtermjs/xterm.js/tree/master/addons/xterm-addon-web-links does that cover everything you need?

**Comment by ivanwonder** [comment]
@jaraco you can try to write a plugin for Hyper with the xtermjs API `registerLinkMatcher`. https://xtermjs.org/docs/api/terminal/classes/terminal/#registerlinkmatcher
maybe can achieve that.

**Comment by jaraco** [comment]
I've never developed on Hyper.js before, so it's going to be a pretty steep ramp-up to learn how to develop on Hyper.js, develop the plugin, and publish it. I'll hope to do that some time in the future when time permits, but I'd welcome someone else to develop that plugin as well too. I'll even offer a US$50 bounty for a solution that's readily configurable on stock Hyper.js via npm (just express interest and I'll post the bounty on bountysource).

**Comment by shamoons** [comment]
> We are now using https://github.com/xtermjs/xterm.js/tree/master/addons/xterm-addon-web-links does that cover everything you need?

Do I need to do anything to enable this? As currently links are not clickable for me without the `hyper-match` plugin

**Comment by Stanzilla** [comment]
no, the problem is that there is no release that includes this. you'd need to self-compile or use a CI build

**Comment by shamoons** [comment]
> no, the problem is that there is no release that includes this. you'd need to self-compile or use a CI build

How can I compile or use a CI build for Win10?

**Comment by Stanzilla** [comment]
https://ci.appveyor.com/project/zeit/hyper/builds/28582621/artifacts
